src/website/utils/url_utils.ts

```diff
@@ -0,0 +1,186 @@
+  // args.fovDownloadHref = datasetData.downloadRoot + "/" + id;
+
+  const fileInfo = await db.getFileInfoByCellId(id);
+  args.imageUrl = baseUrl + fileInfo!.volumeviewerPath;
+  args.parentImageUrl = baseUrl + fileInfo!.fovVolumeviewerPath;
+
+  return args;
+}
+
+export async function getArgsFromParams(urlSearchParams: URLSearchParams): Promise<{
+  args: Partial<AppProps>;
+  viewerSettings: Partial<GlobalViewerSettings>;
+}> {
+  const params = urlSearchParamsToParams(urlSearchParams);
+  let args: Partial<AppProps> = {};
+  const viewerSettings: Partial<GlobalViewerSettings> = {};
+
+  if (params) {
+    if (params.mask) {
+      viewerSettings.maskAlpha = parseInt(params.mask, 10);
+    }
+    if (params.view) {
+      const mapping = {
+        "3D": ViewMode.threeD,
+        Z: ViewMode.xy,
+        Y: ViewMode.xz,
+        X: ViewMode.yz,
+      };
+      const allowedViews = Object.keys(mapping);
+      let view: "3D" | "X" | "Y" | "Z";
+      if (allowedViews.includes(params.view)) {
+        view = params.view as "3D" | "X" | "Y" | "Z";
+      } else {
+        view = "3D";
+      }
+      viewerSettings.viewMode = mapping[view];
+    }
+    if (params.ch) {
+      // ?ch=1,2
+      // ?luts=0,255,0,255
+      // ?colors=ff0000,00ff00
+      const initialChannelSettings: ViewerChannelSettings = {
+        groups: [{ name: "Channels", channels: [] }],
+      };
+      const ch = initialChannelSettings.groups[0].channels;
+
+      const channelsOn = params.ch.split(",").map((numstr) => parseInt(numstr, 10));
+      for (let i = 0; i < channelsOn.length; ++i) {
+        ch.push({ match: channelsOn[i], enabled: true });
+      }
+      // look for luts or color
+      if (params.luts) {
+        const luts = params.luts.split(",");
+        if (luts.length !== ch.length * 2) {
+          console.log("ILL-FORMED QUERYSTRING: luts must have a min/max for each ch");
+        }
+        for (let i = 0; i < ch.length; ++i) {
+          ch[i]["lut"] = [luts[i * 2], luts[i * 2 + 1]];
+        }
+      }
+      if (params.colors) {
+        const colors = params.colors.split(",");
+        if (colors.length !== ch.length) {
+          console.log("ILL-FORMED QUERYSTRING: if colors specified, must have a color for each ch");
+        }
+        for (let i = 0; i < ch.length; ++i) {
+          ch[i]["color"] = colors[i];
+        }
+      }
+      args.viewerChannelSettings = initialChannelSettings;
+    }
+    if (params.url) {
+      const imageUrls = tryDecodeURLList(params.url) ?? decodeURL(params.url);
+      const firstUrl = Array.isArray(imageUrls) ? imageUrls[0] : imageUrls;
+
+      args.cellId = "1";
+      args.imageUrl = imageUrls;
+      // this is invalid for zarr?
+      args.imageDownloadHref = firstUrl;
+      args.parentImageUrl = "";
+      args.parentImageDownloadHref = "";
+      // if json, then use the CFE settings for now.
+      // (See VIEWER_3D_SETTINGS)
+      // otherwise turn the first 3 channels on and group them
```

Comment:
I removed the default `VIEWER_3D_SETTINGS`, @frasercl do you expect this to cause any problems?